chromium_src/ui/views/controls/button/md_text_button.cc

```diff
@@ -39,11 +47,165 @@ class BraveTextButtonHighlightPathGenerator
 
 namespace views {
 
-// To be called from MdTextButtonBase::UpdateColors().
-void MdTextButtonBase::UpdateColorsForBrave() {
+SkColor kBravePrimaryColor = SkColorSetRGB(32, 74, 227);
+MdTextButton::ButtonTheme g_primary_theme = {
+    // Normal Light
+    MdTextButton::ButtonStyle{kBravePrimaryColor, absl::nullopt, SK_ColorWHITE},
+    // Normal Dark
+    MdTextButton::ButtonStyle{kBravePrimaryColor, absl::nullopt, SK_ColorWHITE},
+    // Hover Light
+    MdTextButton::ButtonStyle{SkColorSetRGB(24, 56, 172), absl::nullopt,
+                              SK_ColorWHITE},
+    // Hover Dark
+    MdTextButton::ButtonStyle{SkColorSetRGB(77, 92, 253), absl::nullopt,
+                              SK_ColorWHITE},
+    // Disabled Light
+    MdTextButton::ButtonStyle{SkColorSetA(kBravePrimaryColor, 128),
+                              absl::nullopt, SkColorSetA(SK_ColorWHITE, 128)},
+    // Disabled Dark
+    MdTextButton::ButtonStyle{SkColorSetA(kBravePrimaryColor, 128),
+                              absl::nullopt, SkColorSetA(SK_ColorWHITE, 128)},
+    // Loading Light
+    MdTextButton::ButtonStyle{SkColorSetA(kBravePrimaryColor, 192),
+                              absl::nullopt, SkColorSetA(SK_ColorWHITE, 192)},
+    // Loading Dark
+    MdTextButton::ButtonStyle{SkColorSetA(kBravePrimaryColor, 192),
+                              absl::nullopt, SkColorSetA(SK_ColorWHITE, 192)}};
```

Comment:
b3. Maybe it could be more readable without comments.

```
// consexpr or static base::NoDestructor
base::flat_map<PreferredColorScheme, 
               base::flap_map<ButtonState, ButtonStyle>> g_primary_theme = {
  {PreferedColorScheme::kDark, {
      {ButtonState::kNormal,  
           {.background_color = kBravePrimaryColor,
            .border_color = absl::nullopt,
            .text_color = SK_ColorWHITE}},
        ...
  }}
}

Reply:
If you want to go one more step, you can have type as key to. Then something like this could be done.

```
struct StyleKey {
  Kind kind;
  PreferredColorScheme color_scheme;
  ButtonState state;

  // used for key for ordered map. 
  bool operator<() const {
    return std::tie(kind, color_scheme, state);
  }
}

base::flat_map<StyleKey, Style> {
  {
   {kPrimary, kDark, kNormal}, 
   {.background_color = kBravePrimaryColor, .border_color = absl::nullopt,           
 .text_color = SK_ColorWHITE} }
    ...
   }
}

Then we could remove |theme_| member and access to this global variable directly?

```

Reply:
I've gone with your `StyleKey` approach - it's pretty elegant!